azavea/echo-locator
# Remove voucher number field

## Overview

This PR:
- updates the `UserProfile` data model so that the `voucher_number` field is removed
- adds a migration as a result of the above data model change
- updates the Django backend views, serializers, and admin to remove this field
- updates the housing seeker migration script so that `voucher_number` field is removed and `has_voucher` is set to `True` for all housing seekers
- updates profile edit page on frontend so that the voucher number question is removed

### Checklist


### Notes

There is a `voucherNumber` field on frontend, which is now only used for telling if the mode is anonymous or not. Posting or updating with the field in the payload body is no-op, which is fine.

## Testing Instructions

 * `./scripts/update` and then `./scripts/server`
 * The following items are aimed to test 3 scenarios: existing user, anonymous user, and new user
 * Log in with an existing user account. Go to the profile edit page. Make sure that the voucher number question is gone if you select "Yes" to "Do you have a voucher" question. Make edits to the form field values. Open network traffic tab. Make sure form submission works. Make sure the payload and response of the `/user` endpoint look fine.
 * Log in under anonymous mode. On the profile edit page, fill out the form. Open network traffic tab. Submit the form. Make sure the payload and response of the `/user` endpoint look fine. Go to the profile edit page. Test similarly and make sure it should still work. 
 * Create a new account. In the server logs, copy the link and use it to open the app. On the profile edit page, make sure the form submit correctly. Going back to the profile edit page and editing/submitting the form should also work.
 * If you have ingested user profile from production to your local DB before, go to the Admin UI and remove those users.
 * At the root of the directory, test the following command to make sure that the housing seeker migration script still works:
```
AWS_PROFILE=<your echo profile name> ./scripts/manage migrate_housing_seeker_data -b <production profile bucket name> -p public
```

Resolves https://github.com/azavea/echo-locator/issues/542
